biogtr/io/association_matrix.py

```diff
@@ -22,6 +22,16 @@ class AssociationMatrix:
     ref_instances: list[Instance] = attrs.field()
```

Comment:
These don't need `= attrs.field()` since `auto_attribs=True` is the default in the new attrs API (assuming all fields have type hinting).

![image](https://github.com/talmolab/biogtr/assets/3187454/27619c46-eaae-4bdb-a8b5-3a35449e407b)

https://www.attrs.org/en/stable/api.html#attrs.define